### PR TITLE
Closes #197 197_exprs: replace vars() with exprs()

### LINKED ISSUE #197: General Issue: Change `vars()` to `exprs()`
### Background Information

The admiral core team has decided to change from using `vars()` to `exprs()` (see https://github.com/pharmaverse/admiral/issues/1627 for details).

The admiralonco functions, vignettes, and templates need to be updated accordingly.

- In the vignettes and templates all calls to `vars()` need to be replaced with `exprs()` and `quo_c()` with `expr_c()`.
- In the functions all parameters should expect expressions instead of quosures. For example `new_var <- assert_symbol(enquo(new_var)` should be updated to `new_var <- assert_symbil(enexpr(new_var))`.

### Definition of Done

_No response_

### PR DESCRIPTION
**Pull request by bundfussr**
Please add "Closes #<issue number>" to the title of the pull request. Then the
issue is closed automatically once it is merged to `main`.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralonco/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

The following line requests to update the man pages by the "Man Pages" workflow.

/roxygenize

### CODE REVIEW

**Comment by bundfussr** [comment]
@sgorm123 , @HinalPat , if you want to run any code in this PR you must install the following package versions:
```
remotes::install_github("pharmaverse/admiraldev", ref = "208_exprs@devel")
remotes::install_github("pharmaverse/admiral", ref = "1627_exprs@208_exprs@devel")
```

**Comment by bundfussr** [comment]
@sgorm123 , @HinalPat , could you have a look at this PR?

I have merged the corresponding PRs in admiral and admiraldev today. We need to merge this PR to make admiralonco working with the devel version of admiral and admiraldev.

**Comment by bundfussr** [comment]
As we have merged the pull requests in admiral and admiraldev, you need to call
```
remotes::install_github("pharmaverse/admiraldev", ref = "devel")
remotes::install_github("pharmaverse/admiral", ref = "devel")
```
to install the dependencies now.

**Comment by bundfussr** [comment]
Hi @Thampi-hub , how is your review going? When will you be done with this?

File: R/derive_param_bor.R [hunk 1]
```diff
@@ -121,14 +121,14 @@
...
+#'   new parameter, e.g. `exprs(PARAMCD = "BOR", PARAM = "Best Overall
 #'   Response")` is expected. The values must be symbols, character strings,
 #'   numeric values, or `NA`.
 #'
 #' @param subject_keys Columns to uniquely identify a subject
 #'
-#'   A list of symbols created using `vars()`.
+#'   A list of symbols created using `exprs()`.
 #'
 #'   *Permitted Values:* an `vars` object
```
Comment (Thampi-hub) [inline R/derive_param_bor.R:133]: Should this instead be: an `exprs` object ?

Comment (bundfussr) [reply R/derive_param_bor.R]: Yes, good catch! I have updated it.

**Review by Thampi-hub — APPROVED** [review]
Changes made by Stefan accordingly; Passed review.

**Pull request merged**